Mock-test majority of github_util.GithubClient

kwk opened the pull request:
This removes the necessity for having tokens installed in order to test functions that interact with github.

I've changed some of the methods in `GithubClient` to be `@classmethod`s because they didn't use any internals.

`GithubClient.__label_cache` needed to be made visible for mocking it from the outside.

tuliom commented on snapshot_manager/tests/github_util_test.py:
```diff
@@ -1,125 +1,749 @@
+def test_add_comment_reaction__with_str(github_client_fxt):
+    gh = github_client_fxt
+    node_id = "12345"
+    reaction = github_util.Reaction.LAUGH
```
Why do we need to test this? Does the system react with `LAUGH` anytime?

tuliom replied:
I see you changed line 613, but not line 647 where my comment was added.
Is this intentional?
By the way, I agree line 613 also deserved to be modified.

tuliom commented on snapshot_manager/tests/github_util_test.py:
```diff
@@ -1,125 +1,749 @@
-            body="This comment is for testing TestGithub.test_flip_test_label",
+
+
+def test_add_comment_reaction__unssupprted_type(github_client_fxt):
```
Typo in the name of the function: `unssupprted`

tuliom commented on snapshot_manager/tests/github_util_test.py:
```diff
@@ -1,125 +1,749 @@
+    issue_mock.add_to_labels.assert_called_once_with(new_label)
+    issue_mock.remove_from_labels.assert_called_once_with(
+        "in_testing/fedora-rawhide-x86_64"
+    )
```
Isn't it necessary to check if a single label got removed? Or is this last call doing exactly that?

kwk commented on snapshot_manager/tests/github_util_test.py:
```diff
@@ -1,125 +1,749 @@
+        MyLabel(name="error/rpm__installed_but_unpackaged_files_found", color="FBCA04"),
+        MyLabel(name="error/rpm__file_not_found", color="FBCA04"),
+        MyLabel(name="error/cmake_error", color="FBCA04"),
+        #  MyLabel(name= "error/unknown", color= "FBCA04"),
```
I initially started to add all the `error/` cases that we had the original code and left some out so I can add them later by the test functions below. Leaving the commented out labels here just visualized what other errors we currently do have. Make sense?

tuliom commented on Makefile:
```diff
@@ -1,5 +1,6 @@
 .PHONY: test-snapshot-manager
 test-snapshot-manager:
 	# pytest --doctest-modules
+	coverage erase
```
Suggestion: add a comment mentioning that, e.g.
```
# Ensure previous data won't interfere with the new execution.
```

tuliom commented on snapshot_manager/tests/github_util_test.py:
```diff
@@ -1,125 +1,749 @@
-        cfg.datetime = datetime.datetime.now(datetime.UTC) + datetime.timedelta(days=2)
-        issue = gh.get_todays_github_issue(
-            strategy="big-merge", github_repo="fedora-llvm-team/llvm-snapshots"
+def test_unminimize_comment__unssupprted_type(github_client_fxt):
```
Another typo in the function name `unssupprted`.

kwk replied:
Fixed in a705220

tuliom reviewed (changes requested):
> Hm, you mean the test code or the to-be-tested-code?

I mean all the code in `github_util_test.py` that was added in this PR, e.g. what does `test_create_labels_mix()` do?

> For the record, I've tried to test most of the corner cases and used the naming scheme test_FUNCTION_NAME__CASE_NAME.

I wonder if all functions are following this naming scheme, e.g. `test_create_labels_mix`, `test_remove_labels_safe`.

Thanks!

kwk commented:
@tuliom I've addressed all your comments.

> I wonder if all functions are following this naming scheme, e.g. test_create_labels_mix, test_remove_labels_safe.

I've renamed the `test_create_labels_mix` to `test_create_labels` because it runs through parametrization and each parameter combination triggers a different behaviour. I've kept `test_remove_labels_safe` because there's no special case to separate with `__`. I've also changed the second usage of `LAUGH` to `EYES`.
